### PR TITLE
feat: adds sidecar option to all components

### PR DESCRIPTION
**Pull request by jessesimpson36**
### Which problem does the PR fix?
Related: https://github.com/camunda/camunda-platform-helm/issues/518

This PR introduces sidecar containers to all components except for elasticsearch.

### What's in this PR?
This PR adds a place in values.yaml where a user an input a container spec for extra containers they want deployed alongside a pod.  Also added the extra configuration option to the README.md

### Checklist

Please make sure to follow our [Contributing Guide](../blob/main/CONTRIBUTING.md).

**Before opening the PR:**

- [x] There is no other open [pull request](../pulls) for the same update/change.
- [ ] The commits follow our [Commit Guidelines](../blob/main/CONTRIBUTING.md#commit-guidelines).
- [x] Tests for charts are added (if needed).
- [x] The main Helm chart and sub-chart are updated (if needed).
- [x] In-repo [documentation](../blob/main/CONTRIBUTING.md#documentation) are updated (if needed).

**After opening the PR:**

- [x] Did you sign our CLA (Contributor License Agreement)? It will show once you open the PR.
- [x] Did all checks/tests pass in the PR?

### CODE REVIEW

File: charts/camunda-platform/charts/zeebe-gateway/templates/gateway-deployment.yaml [hunk 1]
```diff
@@ -139,6 +139,9 @@ spec:
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
           {{- end }}
+      {{- if .Values.sidecars}}
```
Comment (aabouzaid) [inline charts/camunda-platform/charts/zeebe-gateway/templates/gateway-deployment.yaml:142]: A space is missing between the key and brackets.

File: charts/camunda-platform/README.md [hunk 2]
```diff
@@ -244,6 +244,7 @@ Check out the default [values.yaml](values.yaml) file, which contains the same c
 | | `identity.keycloak.auth.adminUser` |    Can be used to configure admin user to access existing Keycloak | `""` |
 | | `identity.keycloak.auth.existingSecret` |    Can be used to configure existing Secret object which has admin password to access existing Keycloak | `""` |
 | | `identity.keycloak.auth.existingSecretKey` |    Can be used to configure the key inside existing Secret object which has admin password to access existing Keycloak | `"admin-password"` |
+| | `identity.keycloak.postgresql.primary.sidecars` | Can be set to deploy extra containers in the same postgresql pod | `[]` |
```
Comment (aabouzaid) [inline charts/camunda-platform/README.md:247]: No need to add it for any 3rd party deps.

**Comment by aabouzaid** [comment]
All looks good :rocket: 
I just left a couple of minor comments.

The final thing we need to add is the tests.
Since those values are disabled by default so will not appear in the golden files (the rendered files we use to verify the charts).

Here is the starting point:
https://github.com/camunda/camunda-platform-helm/tree/main/charts/camunda-platform/test/unit

File: charts/camunda-platform/README.md (hunk 2 above)
Comment (jessesimpson36) [reply charts/camunda-platform/README.md:247]: Yeah I think we can expect people to look at the dependency keycloak helm chart if they want to configure that. I'll remove this and the relevant values.yaml section.

File: charts/camunda-platform/charts/zeebe-gateway/templates/gateway-deployment.yaml (hunk 1 above)
Comment (jessesimpson36) [reply charts/camunda-platform/charts/zeebe-gateway/templates/gateway-deployment.yaml:142]: I'll fix this.

File: charts/camunda-platform/values.yaml [hunk 3]
```diff
@@ -191,6 +191,9 @@ zeebe:
     # Image.pullSecrets can be used to configure image pull secrets https://kubernetes.io/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod
     pullSecrets: []
 
+  # Attach extra containers to the zeebe deployment
```
[the remark this answers is not in the document]
Comment (jessesimpson36) [reply charts/camunda-platform/values.yaml:194]: :+1: "Sidecars can be used to attach extra containers to the zeebe deployment"
I can make them all follow this format.

File: charts/camunda-platform/README.md (hunk 2 above)
Comment (jessesimpson36) [reply charts/camunda-platform/README.md]: I removed references to attaching sidecars to keycloak and postgresql.

**Comment by jessesimpson36** [comment]
I've fixed the style/ consistency related issues. Time to write some tests!

**Comment by jessesimpson36** [comment]
Tests are added! exciting stuff!

I'm not sure if I'd recommend integration tests to be added for this feature.  A test I did manually for each service was to portforward to every nginx instance just to ensure that it could be accessed, but any sidecar container is going to be heavily specific to the user who chooses to configure it.

**Comment by aabouzaid** [comment]
@jessesimpson36 all looks good, great job :rocket: 

no need for an integration test for it, the unit test is enough since it's straightforward (if it renders, it works).

I just have a tiny suggestion about the naming of `containersList`, it's in general not recommended to write the type in the var name since it's a statically typed language anyway. So maybe a better name is `podContainers` or even just `containers`.

Apart from that all good :ok_hand:

**Review by aabouzaid — APPROVED** [review]

**Comment by jessesimpson36** [comment]
> I just have a tiny suggestion about the naming of containersList, it's in general not recommended to write the type in the var name since it's a statically typed language anyway. So maybe a better name is podContainers or even just containers.

Good feedback. I also found another issue when I was changing the variable name. The test for the zeebe-gateway referenced Deployment kinds despite being a StatefulSet. It doesn't cause any errors, but I'll fix this.

**Pull request merged**